addons/website/static/src/js/editor/snippets.options.js

```diff
@@ -1680,6 +1713,15 @@ options.registry.topMenuColor = options.Class.extend({
     },
 });
 
+/**
+ * Hide/show footer in the current page.
+ */
+options.registry.HideFooter = VisibilityPageOptionUpdate.extend({
+    pageOptionName: 'footer_visible',
+    showOptionWidgetName: 'hide_footer_page_opt',
+    shownValue: 'shown',
+});
```

Comment:
For this option you duplicated the entire logic you made for the visibility of the header... See how I adapted, the footer code is now only 5 lines. Always try to share code please.